bindgen/src/bindings/cpp/templates/scaffolding/callback.cpp

```diff
@@ -43,19 +43,55 @@ class {{ iface.name() }}Proxy: public {{ iface.name() }} {
                 {% endfor %}
 
                 auto ret = callback_stub(this->handle, {{ loop.index }}, in_buf.data, size, &out_buf);
-
                 rustbuffer_free(in_buf);
 
-                {% match m.return_type() %}
-                {% when Some with (return_type) %}
-                RustStream out_stream(&out_buf);
-                auto result = {{ return_type|read_fn }}(out_stream);
-                rustbuffer_free(out_buf);
+                if (ret == CALL_STATUS_OK) {
+                    {% match m.return_type() %}
+                    {% when Some with (return_type) %}
+                    RustStream out_stream(&out_buf);
+                    auto result = {{ return_type|read_fn }}(out_stream);
+                    rustbuffer_free(out_buf);
+
+                    return result;
+                    {% else %}
+                    rustbuffer_free(out_buf);
+                    {% endmatch %}
+                }
+                else if (ret == CALL_STATUS_ERROR) {
+                    RustStream out_stream(&out_buf);
+                    int32_t v;
+                    out_stream >> v;
+                    {%- if m.throws() %}
+                    switch (v) {
+                    {%- let err_type = m.throws_type().unwrap()|type_name %}
+                    {%- let err_enum = ci.get_enum_definition(err_type).unwrap() %}
+                    {%- for variant in err_enum.variants() %}
+                    {%- let converter_name = err_enum|ffi_converter_name %}
+                    case {{ loop.index }}:
+                    {
+                        auto result = {{ converter_name }}{{ variant.name() }}::read(out_stream, v);
+                        rustbuffer_free(out_buf);
+
+                        throw result;
+                    }
+                    {%- endfor %}
+                    default:
+                        rustbuffer_free(out_buf);
+                        throw std::runtime_error("Unexpected error variant");
+                    }
+                    {%- endif %}
+                    rustbuffer_free(out_buf);
+                    throw std::runtime_error("Callback returned unexpected error code");
```

Comment:
I would also insert the error code that was returned

Reply:
Changed the description here, as if this line triggers, there are no codes to be shown